### PR TITLE
[ENH] Update list of dataframes' categoricals 

### LINKED ISSUE #580: [ENH] Modify dataframe list with union of their categoricals
The problem: vertical `pd.concat([df1, df2, ...])` converts categoricals to strings if the categories do not perfectly overlap between all dataframes. 

The solution: use pandas' `union_categoricals` to modify the original dataframes to add all categories to all dataframes prior to the concatenation.

How to use `union_categoricals`:

From https://stackoverflow.com/questions/45639350/retaining-categorical-dtype-upon-dataframe-concatenation :

```python
from pandas.api.types import union_categoricals

uc = union_categoricals([df1.x,df2.x])
df1.x = pd.Categorical( df1.x, categories=uc.categories )
df2.x = pd.Categorical( df2.x, categories=uc.categories )
```

The final function could iterate over all columns whose names match and are categorical. It can alternatively take an explicit list of particular columns.

### PR DESCRIPTION
**Pull request by zbarry**
# PR Description

Please describe the changes proposed in the pull request: 

- add `unionize_dataframe_categories` & associated tests

image

**This PR resolves #580.**

# PR Checklist

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl
- @hectormz

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -27,6 +28,95 @@
 )
 
 
+def unionize_dataframe_categories(
+    *dataframes, column_names: Union[str, Iterable[str], Any] = None
+) -> List[pd.DataFrame]:
+    """
+    Given a group of dataframes which contain some categorical columns, for
```
Comment (ericmjl) [inline janitor/functions.py]: Minor nitpick on this, @zbarry, but could we start using [semantic line breaks](https://sembr.org) for the docstrings? Yes, there will be more lines, but I think it'll make diffing and editing much easier (as stated in the `sembr` page.

Comment (hectormz) [reply janitor/functions.py]: This is awesome! I never realized I was missing this functionality

Comment (zbarry) [reply janitor/functions.py]: No prob. I think it conforms to the semantic line break protocol now.

**Comment by ericmjl** [comment]
LGTM! :tada:

**Pull request merged**